avalon/tools/widgets.py

```diff
@@ -129,204 +101,68 @@ def get_active_index(self):
         return self.view.currentIndex()
 
     def get_selected_assets(self):
-        """Return the assets' ids that are selected."""
+        """Return the documents of selected assets."""
         selection = self.view.selectionModel()
         rows = selection.selectedRows()
-        return [row.data(self.model.ObjectIdRole) for row in rows]
+        assets = [row.data(self.model.DocumentRole) for row in rows]
 
-    def set_silo(self, silo):
-        """Set the active silo tab"""
-        self.silo.set_current_silo(silo)
+        # NOTE: skip None object assumed they are silo (backwards comp.)
+        return [asset for asset in assets if asset]
 
-    def select_assets(self, assets, expand=True):
-        """Select assets by name.
+    def select_assets(self, assets, expand=True, key="name"):
+        """Select assets by item key.
 
         Args:
-            assets (list): List of asset names
+            assets (list): List of asset values that can be found under
+                specified `key`
             expand (bool): Whether to also expand to the asset in the view
+            key (string): Key that specifies where to look for `assets` values
 
         Returns:
             None
 
+        Default `key` is "name" in that case `assets` should contain single
+        asset name or list of asset names. (It is good idea to use "_id" key
+        instead of name in that case `assets` must contain `ObjectId` object/s)
+        It is expected that each value in `assets` will be found only once.
+        If the filters according to the `key` and `assets` correspond to
+        the more asset, only the first found will be selected.
         """
         # TODO: Instead of individual selection optimize for many assets
 
-        assert isinstance(assets,
-                          (tuple, list)), "Assets must be list or tuple"
+        if not isinstance(assets, (tuple, list)):
+            assets = [assets]
+
+        # convert to list - tuple cant be modified
+        assets = list(assets)
```

Comment:
Looking at the usage of `assets` variable below, would this make more sense to be a `set` for quick lookup?
You might be looking over many many rows for many assets, so it could turn out a slow loop.

It seems `ObjectId` also works inside a `set` so it should be good.